tests/e2e-tests/specs/backend/product-search.test.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * External dependencies
+ */
+import {
+	insertBlock,
+	getEditedPostContent,
+	createNewPost,
+	switchUserToAdmin,
+} from '@wordpress/e2e-test-utils';
+
+describe( 'Product Search', () => {
+	beforeEach( async () => {
+		await switchUserToAdmin();
+		await createNewPost();
+	} );
+
+	it( 'can be created', async () => {
+		await insertBlock( 'Product Search' );
+
+		expect( await getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	it( 'can toggle field label', async () => {
+		await insertBlock( 'Product Search' );
+		await page.click( '.components-form-toggle__input' );
+
+		await expect( page ).not.toMatchElement(
+			'.wc-block-product-search .wc-block-product-search__label'
+		);
+	} );
+
+	it( 'can change field labels in editor', async () => {
+		await insertBlock( 'Product Search' );
+
+		await expect( page ).toFill(
```

Comment:
We can also try to fill this normally, and match it against a snapshot, to also ensure that it saves

Reply:
Can you check again? I think I did what you're after.